File: app/src/main/java/com/zeeshanhussain/timetable/adapters/SimpleFragmentPagerAdapter.java
package com.zeeshanhussain.timetable.adapters;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.zeeshanhussain.timetable.R;
import com.zeeshanhussain.timetable.ui.fragments.FridayFragment;
import com.zeeshanhussain.timetable.ui.fragments.MondayFragment;
import com.zeeshanhussain.timetable.ui.fragments.SaturdayFragment;
import com.zeeshanhussain.timetable.ui.fragments.SundayFragment;
import com.zeeshanhussain.timetable.ui.fragments.ThursdayFragment;
import com.zeeshanhussain.timetable.ui.fragments.TuesdayFragment;
import com.zeeshanhussain.timetable.ui.fragments.WednesdayFragment;

public class SimpleFragmentPagerAdapter extends FragmentPagerAdapter {

    private Context mContext;

    public SimpleFragmentPagerAdapter(FragmentManager fm, Context context) {
        super(fm);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0) {
            return new MondayFragment();
        } else if (position == 1) {
            return new TuesdayFragment();
        } else if (position == 2) {
            return new WednesdayFragment();
        } else if (position == 3) {
            return new ThursdayFragment();
        } else if (position == 4) {
            return new FridayFragment();
        } else if (position == 5) {
            return new SaturdayFragment();
        } else {
            return new SundayFragment();
        }
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }

    @Override
    public int getCount() {
        return 7;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if (position == 0) {
            return mContext.getString(R.string.monday);
        } else if (position == 1) {
            return mContext.getString(R.string.tuesday);
        } else if (position == 2) {
            return mContext.getString(R.string.wednesday);
        } else if (position == 3) {
            return mContext.getString(R.string.thursday);
        } else if (position == 4) {
            return mContext.getString(R.string.friday);
        } else if (position == 5) {
            return mContext.getString(R.string.saturday);
        } else {
            return mContext.getString(R.string.sunday);
        }
    }
}

File: app/src/main/java/com/zeeshanhussain/timetable/adapters/SimpleFragmentPagerAdapter.kt
package com.zeeshanhussain.timetable.adapters

import android.content.Context

import com.zeeshanhussain.timetable.R
import com.zeeshanhussain.timetable.ui.fragments.FridayFragment
import com.zeeshanhussain.timetable.ui.fragments.MondayFragment
import com.zeeshanhussain.timetable.ui.fragments.SaturdayFragment
import com.zeeshanhussain.timetable.ui.fragments.SundayFragment
import com.zeeshanhussain.timetable.ui.fragments.ThursdayFragment
import com.zeeshanhussain.timetable.ui.fragments.TuesdayFragment
import com.zeeshanhussain.timetable.ui.fragments.WednesdayFragment

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.PagerAdapter

class SimpleFragmentPagerAdapter(fm: FragmentManager, private val mContext: Context) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            MondayFragment()
        } else if (position == 1) {
            TuesdayFragment()
        } else if (position == 2) {
            WednesdayFragment()
        } else if (position == 3) {
            ThursdayFragment()
        } else if (position == 4) {
            FridayFragment()
        } else if (position == 5) {
            SaturdayFragment()
        } else {
            SundayFragment()
        }
    }

    override fun getItemPosition(`object`: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun getCount(): Int {
        return 7
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return if (position == 0) {
            mContext.getString(R.string.monday)
        } else if (position == 1) {
            mContext.getString(R.string.tuesday)
        } else if (position == 2) {
            mContext.getString(R.string.wednesday)
        } else if (position == 3) {
            mContext.getString(R.string.thursday)
        } else if (position == 4) {
            mContext.getString(R.string.friday)
        } else if (position == 5) {
            mContext.getString(R.string.saturday)
        } else {
            mContext.getString(R.string.sunday)
        }
    }
}